gradio/routes.py

```diff
@@ -265,7 +265,12 @@ async def file(path: str, request: fastapi.Request):
             created_by_app = str(utils.abspath(path)) in set().union(
                 *blocks.temp_file_sets
             )
-            if in_app_dir or created_by_app:
+            in_file_dir = any(
+                dir in (str(p) for p in utils.abspath(path).parents)
+                for dir in blocks.file_directories
+            )
```

Comment:
Can we do it this way to take advantage of pathlib's built-in `in` functionality
```suggestion
            in_file_dir = any((utils.abspath(dir) in utils.abspath(path).parents for dir in blocks.file_directories))
```